src/main/java/InterfaceAdapters/ChatGateway.java

```diff
@@ -50,6 +51,8 @@ public void saveToFile(String filepath, Object chatRepoUseCase) throws IOExcepti
         OutputStream file = new FileOutputStream(filepath);
         OutputStream buffer = new BufferedOutputStream(file);
         ObjectOutput output = new ObjectOutputStream(buffer);
+        ChatRepoUseCase chatRepoUseCase1 = (ChatRepoUseCase) chatRepoUseCase;
+        chatRepoUseCase1.savingChats();
```

Comment:
Static variables cannot be serialized, so must convert currChatrooms to an instance attribut before you can serialize chatRepoUseCase